src/app/application.cpp

```diff
@@ -278,7 +279,7 @@ void Application::torrentFinished(BitTorrent::TorrentHandle *const torrent)
         program.replace("%T", torrent->currentTracker());
         program.replace("%I", torrent->hash());
 
-        QProcess::startDetached(program);
+        std::thread ([program]() { return system(program.toStdString().c_str()); }).detach();
```

Comment:
@glassez 
>process waits until all its threads finished.

That only happens when you `join()` the thread.

Add a 10sec delay in `system()` (in windows):
`system("PING 1.1.1.1 -n 1 -w 10000 >NUL & copy c:\\whatever.iso d:\\");`
the process ends first then comes the detached thread.